tests/core/offers/test_api.py

```diff
@@ -1137,3 +1137,11 @@ def test_cannot_update_pending_offer_validation_with_a_rejected_offer(self):
 
         assert is_offer_updated is False
         assert offer.validation == OfferValidationStatus.REJECTED
+
+    @mock.patch("pcapi.connectors.redis.add_offer_id")
+    def test_update_pending_offer_validation_status_and_reindex_in_algolia(self, mocked_add_offer_id):
+        offer = OfferFactory(validation=OfferValidationStatus.PENDING)
+
+        update_pending_offer_validation_status(offer, OfferValidationStatus.APPROVED)
+
+        mocked_add_offer_id.assert_any_call(client=app.redis_client, offer_id=offer.id)
```

Comment:
```suggestion
        mocked_add_offer_id.assert_called_once_with(client=app.redis_client, offer_id=offer.id)
```

On pourrait vérifier ainsi que l'enregistrement dans le cache redis est appelé une seule fois.